### PR TITLE
feat(spans): Add a new recombiner consumer

### PR DESCRIPTION
**Pull request by shruthilayaj**
We're moving away from celery tasks + rabbit and instead
want a second consumer that processes segments. This
PR adds the new topic and consumer and refactors code
away from tasks. 

This code doesn't run in prod yet.

In a follow up, the `process-spans` consumer will fetch
segments that are ready to be processed and push them
to the `buffered-segments` topic

### CODE REVIEW

**Comment by shruthilayaj** [comment]
Ah I need to define a schema for the new topic

**Comment by shruthilayaj** [comment]
Blocked by https://github.com/getsentry/sentry-kafka-schemas/pull/237

File: src/sentry/spans/consumers/recombine/factory.py
Comment (shruthilayaj) [inline src/sentry/spans/consumers/recombine/factory.py:21]: Will update this to validate schema once it's available, probably as a follow up

File: src/sentry/conf/types/kafka_definition.py
Comment (evanh) [inline src/sentry/conf/types/kafka_definition.py:48]: Does this topic actually exist in prod?

File: src/sentry/spans/consumers/recombine/factory.py
Comment (evanh) [inline src/sentry/spans/consumers/recombine/factory.py:29]: Why sample this?

File: src/sentry/utils/sdk.py
Comment (antonpirker) [inline src/sentry/utils/sdk.py:41]: Why do those have a `/` prefixed in contrast to the others?

**Review by antonpirker — COMMENTED** [review]
Just left one question.

File: src/sentry/conf/types/kafka_definition.py
Comment (shruthilayaj) [reply src/sentry/conf/types/kafka_definition.py]: No, it does not. None of these consumers/topics/redis existing in prod yet

File: src/sentry/spans/consumers/recombine/factory.py
Comment (shruthilayaj) [reply src/sentry/spans/consumers/recombine/factory.py]: Updated to log exception instead - had just copied over an old pattern

File: tests/sentry/spans/consumers/process/test_factory.py
Comment (shruthilayaj) [inline tests/sentry/spans/consumers/process/test_factory.py:18]: Just plopped in a different span here, sorry!

File: src/sentry/conf/types/kafka_definition.py
Comment (evanh) [reply src/sentry/conf/types/kafka_definition.py]: Is there a concern of deploying this when the infrastructure doesn't exist?

Comment (shruthilayaj) [reply src/sentry/conf/types/kafka_definition.py]: The consumer code is for sure okay to deploy without the infra in place (since ops actually needs to manually deploy the consumers), and the consumers producing to and reading from this topic aren't actually running in production, so I assumed it would be fine. @phacops do you know if that's the case?

Comment (shruthilayaj) [reply src/sentry/conf/types/kafka_definition.py]: @evanh I'm going to make an ops ticket to get this topic configured and make sure that goes through before I merge this PR just to be safe

Comment (shruthilayaj) [reply src/sentry/conf/types/kafka_definition.py]: Ops [ticket](https://getsentry.atlassian.net/browse/OPS-5494)

**Review by phacops — APPROVED** [review]

File: src/sentry/conf/types/kafka_definition.py
Comment (shruthilayaj) [reply src/sentry/conf/types/kafka_definition.py]: The preference is to merge all the code first and then deploy infra to prod in the right order (details in the ticket), so I'm going to merge this PR cc: @evanh

**Pull request merged**
